### ISSUE TITLE
Enhance `Modal` / `Flyout` component to support `alertdialog` role

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#3337]
Scope
---
The `Modal` / ~~`Flyout`~~ component is intended for different purposes but is missing acting like an `alertdialog` role from a semantic point of view. For announcing critical information inside a Modal Dialog to assistive technologies, the component should support the ARIA `role="alertdialog"`.

Notes  
---
Example: https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/examples/alertdialog/

Acceptance criteria
---
- [x] `Modal` / ~~`Flyout`~~ component should be configurable to expose an `alertdialog` role to assistive technologies

Subtasks
---
- [ ] Task
